src/ui/prompts/commonPrompts.ts

```diff
@@ -36,3 +39,29 @@ export const descriptionPrompt: Prompt = {
     transformer: hintTextTransformer('(Optional)'),
     type: 'input'
 }
+
+export const transformResponse = (answers: Record<string, unknown>, prompts: Prompt[]) => {
+    const result = answers
+    prompts.forEach((prompt) => {
+        if (result[prompt.name] && prompt.transformResponse) {
+            result[prompt.name] = prompt.transformResponse(result[prompt.name])
+        }
+    })
+    return result
+}
+
+export const handleCustomPrompts = async (prompts: Prompt[], authToken: string, projectKey: string) => {
+    const [listOptionsPrompts, standardPrompts] = partition(prompts, (prompt) => !!prompt.listOptionsPrompt)
+
+    const result = await inquirer.prompt(standardPrompts, {
+        token: authToken,
+        projectKey: projectKey
+    })
+    for (const prompt of listOptionsPrompts) {
+        if (prompt.listOptionsPrompt) {
```

Comment:
whats the reason for this check? Arent the prompts partitioned by the existence of this field?

Reply:
good point, the problem was `partition` doesn't narrow the type so typescript was complaining that listOptionsPrompt could still be undefined here but i can get around it by casting it like: 
```
for (const prompt of listOptionsPrompts as ListPrompt[]) {
        result[prompt.name] = await prompt.listOptionsPrompt()
    }
```

Reply:
yeah that seems fine to me